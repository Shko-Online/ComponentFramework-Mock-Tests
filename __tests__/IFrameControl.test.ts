/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

import * as sinon from "sinon";

import {
  ComponentFrameworkMockGenerator,
  StringPropertyMock,
  DecimalNumberPropertyMock,
} from "@shko.online/componentframework-mock";
import { IFrameControl } from "@powerapps-samples/i-frame-control/IFrameControl";
import {
  IInputs,
  IOutputs,
} from "@powerapps-samples/i-frame-control/IFrameControl/generated/ManifestTypes";
import resource from "@powerapps-samples/i-frame-control/IFrameControl/strings/IFrameControl.1033.resx";

describe("IFrameControl", () => {
  let mockGenerator: ComponentFrameworkMockGenerator<IInputs, IOutputs>;
  beforeEach(() => {
    window.alert = function (s: string) {
      console.info(s);
    };
    const container = document.createElement("div");

    mockGenerator = new ComponentFrameworkMockGenerator(
      IFrameControl,
      {
        stringProperty: StringPropertyMock,
        latitudeValue: DecimalNumberPropertyMock,
        longitudeValue: DecimalNumberPropertyMock,
      },
      container
    );
    mockGenerator.SetControlResource(resource);
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.innerHTML = null;
  });
  it("Init should work", () => {
    mockGenerator.ExecuteInit();
    sinon.assert.calledOnce(mockGenerator.control.init);
    expect(document.body).toMatchSnapshot();
  });
  it("Update View should Work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    sinon.assert.calledOnce(mockGenerator.control.init);
    sinon.assert.calledOnce(mockGenerator.control.updateView);
    expect(document.body).toMatchSnapshot();
  });
  it("Update View with value should work", () => {
    mockGenerator.metadata.initItems({
      "@odata.context": "#!CanvasApp",
      value: [
        {
          latitudeValue: 41.153332,
          longitudeValue: 20.168331,
        },
      ],
    });
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    sinon.assert.calledOnce(mockGenerator.control.init);
    sinon.assert.calledOnce(mockGenerator.control.updateView);
    expect(document.body).toMatchSnapshot();
  });
});
