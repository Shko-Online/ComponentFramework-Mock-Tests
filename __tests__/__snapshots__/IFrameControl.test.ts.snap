// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IFrameControl Init should work 1`] = `
<body>
  <div />
</body>
`;

exports[`IFrameControl Update View should Work 1`] = `
<body>
  <div>
    <iframe
      class="SampleControl_IFrame"
    />
  </div>
</body>
`;

exports[`IFrameControl Update View with value should work 1`] = `
<body>
  <div>
    <iframe
      class="SampleControl_IFrame"
      src="https://www.bing.com/maps/embed?h=400&w=300&cp=41.153332~20.168331&lvl=12&typ=d&sty=o&src=SHELL&FORM=MBEDV8"
    />
  </div>
</body>
`;
